### PR TITLE
update types/node to latest

### PR DESCRIPTION
**Pull request by smartiniOnGitHub**
update types/node to latest, and all other dependencies to latest; do a small fix for a failing test (since this upgrade)

Related issue is in Fastify/Help, [here](https://github.com/fastify/help/issues/57) .

Thanks,
Sandro

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/types/index.ts [hunk 1]
```diff
@@ -417,7 +417,7 @@ server.setErrorHandler((err, request, reply) => {
 server.listen(3000, err => {
   if (err) throw err
   const address = server.server.address()
-  if (typeof address === 'object') {
+  if (typeof address === 'object' && address !== null) {
```
Comment (cemremengu) [inline test/types/index.ts]: Why this change? I think if it is null, the check will fail here anyways, right?

Comment (Ethan-Arrowood) [reply test/types/index.ts]: typeof null returns 'object'

Comment (Ethan-Arrowood) [reply test/types/index.ts]: @smartiniOnGitHub i think a better check here would be to use short-circuiting:
```js
if (address && typeof address === 'object')
```

Comment (cemremengu) [reply test/types/index.ts]: TIL. Thanks @Ethan-Arrowood !

Comment (smartiniOnGitHub) [reply test/types/index.ts]: Hi, without that change the test fails; but not sure if this is the right fix ...

**Comment by smartiniOnGitHub** [comment]
@cemremengu  @Ethan-Arrowood @mcollina now I should do last change as proposed or did you already applied ? Thanks and sorry for the noise ...

File: test/types/index.ts (hunk 1 above)
Comment (smartiniOnGitHub) [reply test/types/index.ts]: @Ethan-Arrowood Changes applied, all tests good even now. Thanks for the help :-)

**Comment by smartiniOnGitHub** [comment]
Sorry I have another small change here: add /*.tgz and /*.zip to .gitignore (to avoid commit npm generated package files and zip files in project root), and add .nyc_output and coverage/ to .npmignore (to avoid add these test coverage folders when generating npm packages) ... do you think it could be a good addition ? If yes here, so it will be easier for you to merge with this code ?
Tell me ... thanks a lot. Bye, Sandro

**Comment by mcollina** [comment]
> Sorry I have another small change here: add /_.tgz and /_.zip to .gitignore (to avoid commit npm generated package files and zip files in project root), and add .nyc_output and coverage/ to .npmignore (to avoid add these test coverage folders when generating npm packages) ... do you think it could be a good addition ? If yes here, so it will be easier for you to merge with this code ?
Tell me ... thanks a lot. Bye, Sandro

Please send a separate PR!
